Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/qrcode/PosterTemplateProvider.kt

```diff
@@ -0,0 +1,64 @@
+package de.rki.coronawarnapp.eventregistration.checkins.qrcode
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.pdf.PdfRenderer
+import android.os.ParcelFileDescriptor
+import de.rki.coronawarnapp.eventregistration.events.server.qrcodepostertemplate.QrCodePosterTemplateServer
+import de.rki.coronawarnapp.server.protocols.internal.pt.QrCodePosterTemplate.QRCodePosterTemplateAndroid.QRCodeTextBoxAndroid
+import de.rki.coronawarnapp.util.di.AppContext
+import timber.log.Timber
+import java.io.File
+import java.io.FileOutputStream
+import javax.inject.Inject
+import kotlin.math.roundToInt
+
+class PosterTemplateProvider @Inject constructor(
+    private val posterTemplateServer: QrCodePosterTemplateServer,
+    @AppContext private val context: Context
+) {
+    @Suppress("BlockingMethodInNonBlockingContext")
+    suspend fun template(): Template {
```

Comment:
I think we should run this suspend function `withContext(Dispatchers.IO)`. I know, it is currently already called from IO, but it's best practice for suspend functions to be callable from the main thread without blocking.

Reply:
this is a utility class ,so it is the responsibility of the called to do that , which is already happening

Reply:
I am still the opinion, that its best practise to use `withContext(Dispatchers.IO)` here, as you e.g. can read [here](https://elizarov.medium.com/blocking-threads-suspending-coroutines-d33e11bf4761) - Section "Suspending Convention".

This convention shifts the responsibility about the Thread which is used from the caller to the suspend function itself.

Reply:
Addressed